pymongo/auth.py

```diff
@@ -13,15 +13,17 @@
 # limitations under the License.
 
 """Authentication helpers."""
+from __future__ import annotations
```

Comment:
I like this. What do you think about opening a new ticket to add this import to all files so we don't have to deal with forward references as strings (like we do in SON)?

Reply:
I used autotyping to start, but even with the --aggressive flag it didn't do a ton beyond the very simple functions, so most of it was by hand. 

Adding that import everywhere is a good idea, I'll file a ticket.